packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +57,66 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const calculateTimezoneByValue = (timeZone, date = new Date()): number | undefined => {
+  if (!timeZone) {
+    return undefined;
+  }
+
+  let offset;
+  try {
+    offset = getStringOffset(timeZone, date);
+  } catch (e) {
+    return undefined;
+  }
+
+  if (offset === 'GMT') {
+    return 0;
   }
-  return timezone;
+
+  const isMinus = offset.substring(3, 4) === '-';
+  const hours = offset.substring(4, 6);
+  const minutes = offset.substring(7, 9);
+
+  const result = parseInt(hours, 10) + parseInt(minutes, 10) / 60;
+  return isMinus ? -result : result;
+};
+
+// 'GMT±XX:YY' or 'GMT' format
+const getStringOffset = (timeZone, date = new Date()): string => {
+  const dateTimeFormat = new Intl.DateTimeFormat('en-US', {
+    timeZone,
+    timeZoneName: 'longOffset',
+  } as any);
+
+  return dateTimeFormat.formatToParts(date)
+    .filter((part) => part.type === 'timeZoneName')
+    .map((p) => p.value)[0];
```

Comment:
Let's refactor this part of the code and make it more safe:
```suggestion
  return dateTimeFormat.formatToParts(date)?.find((part) => part.type === 'timeZoneName')?.value
```

Reply:
[formatToParts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts) returns the Array as a result, not null or undefined.

Similarly, the formatting _timeZoneName: 'longOffset'_ splits the date into parts, one of which is _'timeZoneName'_. If this is not the case, the component will not work, and it won't be possible to determine why. I think there should be exceptions here as well.

Why do you want use optional chaining operator?

Reply:
We agreed to throw an exception here